src/main/java/io/airlift/units/DataSize.java

```diff
@@ -115,8 +115,9 @@ public long toBytes()
     }
 
     /**
-     * Prefer {@link DataSize#toBytes()} since it avoids conversion to a floating point representation
+     * @deprecated Use {@link DataSize#toBytes()} instead to avoid floating point precision semantics
```

Comment:
I'm not sure deprecating this make sense, as it basically makes the stored unit useless. Why have `to(Unit)` at all in that case (since you can't get the fractional value in that unit)?

Reply:
The unit is useful for printing (i.e., `toString`)

Reply:
Indeed. I marked all methods that expose the fractional semantics as deprecated instead of outright removing them. This method clearly falls in that category. The preservation of unit (except when serialized and deserialized as JSON) is kept since printing config values in the same format  as they were specified has some value, but accessing the fractional value should be discouraged except in the toString() context.